server/aesdsocket.c

```diff
@@ -202,34 +277,41 @@ int main(int argc, char **argv)
 		// Receive and write to file
 		fd =open(WRITE_FILE,O_WRONLY);
 	        lseek(fd,0,SEEK_END);
+	        unsigned long index = 0;
 		while(1){
 		
 			ssize_t rBytes = recv(clientfd, buffer, sizeof(buffer), 0);
-			//printf("Received %ld bytes\n",rBytes);
-			write(fd, buffer, rBytes);
+			rdBuff = realloc(rdBuff,index+rBytes);
+			memcpy(rdBuff+index,buffer,rBytes);
+			index += rBytes;
 			total_b += rBytes;
-			if(buffer[rBytes-1] == '\n')
+			if(buffer[rBytes-1] == '\n'){
+				write(fd, rdBuff, index);
 				break;
+			}
 		}
-		close(fd);		
+		close(fd);	
+		free(rdBuff);
+		rdBuff=NULL;	
 		
-		//printf("%ld\n",total_b);
 		// Send 
 		fd =open(WRITE_FILE,O_RDONLY);
 		ssize_t rd = total_b;
+		rdBuff = realloc(rdBuff,rd);
```

Comment:
@ajsanthosh14   what assumption is made here about the size of RAM heap relative to the total size of all packets sent?